cmd/snap/cmd_info.go

```diff
@@ -417,6 +417,17 @@ func (iw *infoWriter) printDescr() {
 	printDescr(iw, iw.theSnap.Description, iw.termWidth)
 }
 
+func (iw *infoWriter) maybePrintCategories() {
+	if !iw.verbose || len(iw.theSnap.Categories) == 0 {
+		return
+	}
+
+	fmt.Fprintf(iw, "categories:\n")
```

Comment:
for symmetry it would be good if `snap find` took --category now in favor of the old --section, same for the related APIs, though we need for now to support both

Reply:
I had a look at implementing this but this change would be substantial. I suggest this is done in a separate PR if required (it's not required for the desktop store at this time).

Reply:
it's messy for the API to take section as input in one place and return categories in another

Reply:
I've dropped the output from snap info - it's not required for this change.

Reply:
that doesn't really change the API level issue though

Reply:
I've added support /v2/find?category=foo and /v2/categories to get the list.